chia/rpc/wallet_rpc_api.py

```diff
@@ -1587,7 +1624,8 @@ async def nft_get_by_did(self, request) -> EndpointResult:
 
     async def nft_get_wallet_did(self, request) -> EndpointResult:
         wallet_id = uint32(request["wallet_id"])
-        nft_wallet: NFTWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        nft_wallet = self.service.wallet_state_manager.wallets[wallet_id]
+        assert isinstance(nft_wallet, NFTWallet)
```

Comment:
I do here-by amend my review.  We can't be asserting on external input.  This should be an error like `requested wallet id <ID> is a <wallet type> not the needed nft wallet type`.  Or...  something.

Looks like there may be a few of these to address.

This came up in support based on a response of `Failed to mint NFT: {'error': '', 'success': False}` and log traceback.

```python-traceback
wallet chia.rpc.util : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia/rpc/util.py", line 18, in inner
  File "chia/rpc/wallet_rpc_api.py", line 1743, in nft_get_wallet_did
AssertionError
```

Reply:
Exploring in https://github.com/Chia-Network/chia-blockchain/pull/14030.